src/mesh/refinement_cc_in_one.cpp

```diff
@@ -218,5 +223,218 @@ void ComputePhysicalRestrictBounds(MeshData<Real> *md) {
   Kokkos::Profiling::popRegion(); // ComputePhysicalRestrictBounds_MeshData
 }
 
+void Prolongate(cell_centered_bvars::BufferCache_t &info, IndexShape &cellbounds,
+                IndexShape &c_cellbounds) {
+  const IndexDomain interior = IndexDomain::interior;
+  const IndexDomain entire = IndexDomain::entire;
+  auto ckb = c_cellbounds.GetBoundsK(interior);
+  auto cjb = c_cellbounds.GetBoundsJ(interior);
+  auto cib = c_cellbounds.GetBoundsI(interior);
+  auto kb = cellbounds.GetBoundsK(interior);
+  auto jb = cellbounds.GetBoundsJ(interior);
+  auto ib = cellbounds.GetBoundsI(interior);
+
+  const int nbuffers = info.extent_int(0);
+  const int scratch_level = 1; // 0 is actual scratch (tiny); 1 is HBM
+  size_t scratch_size_in_bytes = 1;
+
+  if (cellbounds.ncellsk(entire) > 1) { // 3D
```

Comment:
Can this be re-written so that there is not a separate piece of code for each dimensionality? I have not looked in detail, but it appears that there are just extra terms that show up as you go up in dimensions. One possibility is to make this a separate template function with the dimension as a template parameter and then just have `if` statements in a single loop. I can imagine having one piece of code to maintain will make implementing other prolongation methods easier.

Reply:
Can't you just add 
```c++
int sk = info(buf).sk;
int ek = info(buf).ek;
if (ndim < 3) {
  sk = ckb.s
  ek = ckb.s
}
int sj = info(buf).sj;
int ej = info(buf).ej;
if (ndim < 2) {
  sj = cjb.s
  ej = cjb.s
}
```
and then loop over 
```c++
par_for_inner(inner_loop_pattern_ttr_tag, team_member, 0, info(buf).Nt - 1, 0,
                          info(buf).Nu - 1, 0, info(buf).Nv - 1, sk, ek, sj, ej, 
                          info(buf).si, info(buf).ei,
                          [&](const int l, const int m, const int n, const int k,
                              const int j, const int i) {
```
which will give `int fk = (k - ckb.s) * 2 + kb.s = kb.s`, which is what is set by hand now at lower dimensionality.

Reply:
@lroberts36 take a look now.